comm/src/main/scala/coop/rchain/comm/transport/StreamObservable.scala

```diff
@@ -31,15 +34,20 @@ class StreamObservable(bufferSize: Int, folder: Path)(implicit log: Log[Task], s
     def push(file: Path): Task[Boolean] =
       Task.delay(subject.pushNext(StreamToPeers(peers, file, blob.sender)))
 
-    def propose(file: Path): Task[Unit] =
-      push(file) >>= (_.fold(Task.unit, retry(file)))
+    def propose(file: Path, retryCount: Int): Task[Unit] =
+      push(file) >>= (_.fold(Task.unit, retry(file, retryCount)))
 
-    def retry(file: Path): Task[Unit] =
+    def retry(file: Path, retryCount: Int): Task[Unit] =
       Task
-        .defer(log.warn("Retrying push to client stream") >> propose(file))
-        .delayExecution(1.second)
+        .defer(logRetry(retryCount) >> propose(file, retryCount + 1))
+        .delayExecution(RetryDelay)
+
+    def logRetry(retryCount: Int): Task[Unit] =
+      if (retryCount % LogRetryEvery == 0)
+        log.warn(s"Retrying push to client stream. Retry $retryCount")
```

Comment:
Information does not say anything about queue being full. Maybe it's worth mentioning.